saga/Repository.py

```diff
@@ -381,4 +385,113 @@ def _copy_file_to_dir(self, src, dst):
         if not os.path.isdir(dirname):
             os.makedirs(dirname)
 
-        shutil.copyfile(src, join(dst, src))
\ No newline at end of file
+        shutil.copyfile(src, join(dst, src))
+
+
+
+    def pull_folder(self, relative_folder_path):
+        # api-endpoint 
+        URL = "http://localhost:3000/get-folder"
+
+        base_directory = self.base_directory
+
+        shutil.rmtree(self.saga_directory)
+                
+        if not base_directory[len(base_directory) -1] == '/':
+            base_directory = base_directory + "/"
+        
+        try:
+            response = requests.get(url = URL, data={'folder_location' : relative_folder_path})
+            paths = response.json()
+            
+            # download all of the files
+            for path in paths['file_paths']:
+                self._pull_file(path)
+
+            # make all of the empty folders
+            for path in paths['empty_folder_paths']:
+                empty_folder_location = base_directory + path
```

Comment:
Check out the `os.path.join` function. Everywhere you add paths together (and mess about with "/" on the end and not), this should be used.